Assets/Scripts/Combat/TargetSelector.cs

```diff
@@ -53,33 +56,71 @@ public void Target(Ability callingAbility, bool userTargeting = true)
         {
             // TODO do something else, add rest of options
             if (CurrentTargetSchema.CombatantType == CombatantType.Enemy)
-                TargetAllEnemies(userTargeting);
+                StartCoroutine(TargetSingleEnemy());
             else if (CurrentTargetSchema.CombatantType == CombatantType.Ally)
                 TargetAllAllies(userTargeting);
         }
     }
 
-    private void TargetSingleEnemy(bool userTargeting = true)
+    private IEnumerator TargetSingleEnemy()
     {
-        // TODO
+        MouseSelector.enabled = true;
+        MouseSelector.IsSingleTargetting = true;
+        
+        while (!MouseSelector.IsTargetSelected)
+        {
+            if (MouseSelector.IsRegrettingDecision)
+            {
+                MouseSelector.enabled = false;
+                Debug.Log("Regretting decision");
+                MouseSelector.IsRegrettingDecision = false;
+                CombatSystem.GoBackToAbilitySelect();
+                yield break;
+            }
+            yield return null;
+        }
+        
+        MouseSelector.IsSingleTargetting = false;
+        MouseSelector.IsTargetSelected = false;
+        MouseSelector.enabled = false;
+
+        ReplyToCallingAbility();
+    }
+
+    public void ChooseTarget(GameObject target)
+    {
+        GameObject[] enemy = {target};
+        CurrentTargetedCombatants = enemy;
```

Comment:
actually, since you don't do any operations on `enemy`, just use `CurrentTargettedCombatants = { target }` in one shot

Reply:
Tried it, doesn't work